native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBResultParameterProcessor.java

```diff
@@ -163,6 +166,47 @@ public Object processCustomTypeFromResultSet(ResultSet resultSet, int columnInde
         }
     }
 
+    @Override
+    public BArray convertArray(Array array, int sqlType, Type type) throws SQLException, DataError {
+        if (array != null) {
+            Utils.validatedInvalidFieldAssignment(sqlType, type, "SQL Array");
+            Object[] dataArray = (Object[]) array.getArray();
+            if (dataArray == null || dataArray.length == 0) {
+                return null;
+            }
+            Object firstNonNullElement = firstNonNullObject(dataArray);
+            boolean containsNull = containsNullObject(dataArray);
+            return createAndPopulateVArrays(firstNonNullElement, dataArray, type, array, containsNull);
+        } else {
+            return null;
+        }
+    }
+
+    private BArray createAndPopulateVArrays(Object firstNonNullElement, Object[] dataArray,
+                                                          Type type, Array array, Boolean containsNull)
```

Comment:
```suggestion
    private BArray createAndPopulateVArrays(Object firstNonNullElement, Object[] dataArray,
                                                                            Type type, Array array, Boolean containsNull)
```

Reply:
corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/28/commits/2ed0ab7c618c846dd6133b3a7385ddea63c04b8d